be/test/storage/sstable/persistent_index_sstable_test.cpp

```diff
@@ -211,37 +211,42 @@ TEST_F(PersistentIndexSstableTest, test_persistent_index_sstable) {
     // 1. build sstable
     const std::string filename = "test_persistent_index_sstable_1.sst";
     ASSIGN_OR_ABORT(auto file, fs::new_writable_file(lake::join_path(kTestDir, filename)));
-    phmap::btree_map<std::string, sstable::IndexValueWithVer, std::less<>> map;
+    phmap::btree_map<std::string, std::list<sstable::IndexValueWithVer>, std::less<>> map;
     for (int i = 0; i < N; i++) {
-        map.insert({fmt::format("test_key_{:016X}", i), std::make_pair(100, i)});
+        std::list<sstable::IndexValueWithVer> index_value_vers;
+        index_value_vers.emplace_front(100, i);
+        map.insert({fmt::format("test_key_{:016X}", i), index_value_vers});
     }
-    uint64_t filesz = 0;
-    ASSERT_OK(sstable::PersistentIndexSstable::build_sstable(map, file.get(), &filesz));
+    uint64_t filesize = 0;
+    ASSERT_OK(sstable::PersistentIndexSstable::build_sstable(map, file.get(), &filesize));
     // 2. open sstable
     std::unique_ptr<sstable::PersistentIndexSstable> sst = std::make_unique<sstable::PersistentIndexSstable>();
     ASSIGN_OR_ABORT(auto read_file, fs::new_random_access_file(lake::join_path(kTestDir, filename)));
     std::unique_ptr<Cache> cache_ptr;
     cache_ptr.reset(new_lru_cache(100));
-    ASSERT_OK(sst->init(read_file.get(), filesz, cache_ptr.get()));
+    PersistentIndexSstablePB sstable_pb;
+    sstable_pb.set_filename(filename);
+    sstable_pb.set_filesize(filesize);
+    ASSERT_OK(sst->init(std::move(read_file), sstable_pb, cache_ptr.get()));
 
     {
         // 3. multi get with version (all keys included)
         std::vector<std::string> keys_str(N / 2);
         std::vector<Slice> keys(N / 2);
         std::vector<IndexValue> values(N / 2, IndexValue(NullIndexValue));
         std::vector<IndexValue> expected_values(N / 2);
-        sstable::KeyIndexesInfo key_indexes_info;
-        sstable::KeyIndexesInfo found_keys_info;
+        std::set<uint32_t> key_indexes_info;
+        std::set<uint32_t> found_keys_info;
         for (int i = 0; i < N / 2; i++) {
             int r = rand() % N;
             keys_str[i] = fmt::format("test_key_{:016X}", r);
             keys[i] = Slice(keys_str[i]);
             expected_values[i] = r;
-            key_indexes_info.key_index_infos.push_back(i);
+            key_indexes_info.insert(i);
```

Comment:
is it a design assumption that the key_indexes_info, are always [0, 1, 2, ....]?